src/shared/internal-calendar/internal-calendar-day.style.tsx

```diff
@@ -5,6 +5,7 @@ import { DayVariant } from "./internal-calendar-day";
 
 export interface StyleProps {
     $disabled?: boolean;
+    $disallowed?: boolean;
```

Comment:
this could be confusing for someone coming in for the first time. Perhaps `$disabledDisplayOnly` or `$disabledDisplay` might be clearer

same goes to the object key in the status

Reply:
since `TimeSlotWeekView` also has some interaction management, I reworked these two style props to be `$interactive` which manages hover effects and `$disabledDisplay` which handles disabled styling